js/article.js

```diff
@@ -9,28 +6,77 @@ function Article(blogObject) {
   this.image = blogObject.image;
   this.caption = blogObject.caption;
   this.body = blogObject.body;
-  this.blogDataToHtml = function() {
-    var template = Handlebars.compile($('#template').html());
-    var date = new Date(this.publishedOn);
-    // TODO:  if published today, display today
-    this.publishedLongDate = date.toString("dddd, MMMM d, yyyy");
-    this.daysAgo = parseInt((new Date() - date)/60/60/24/1000);
-    this.publishStatus = this.publishedOn ? `<p>${this.publishedLongDate}   <em>(${this.daysAgo} days ago</em>)</p>` : '<p>(draft)</p>';
-    return template(this);
-  }
 }
 
-// run the blog data through the constructor function and add to the array
-blogData.forEach(function(dataObject) {
-  articles.push(new Article(dataObject));
-});
+// array that will hold all the blog article objects
+Article.all = [];
+
+Article.prototype.blogDataToHtml = function() {
+  var template = Handlebars.compile($('#template').html());
+  var date = new Date(this.publishedOn);
+  // TODO:  if published today, display today
+  this.publishedLongDate = date.toString("dddd, MMMM d, yyyy");
+  this.daysAgo = parseInt((new Date() - date)/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? `<p>${this.publishedLongDate}   <em>(${this.daysAgo} days ago</em>)</p>` : '<p>(draft)</p>';
+  return template(this);
+}
 
-// sort the articles so the newest is on top
-articles.sort(function(a,b) {
-  return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-});
+// output to console to get valid json
+// console.log(JSON.stringify(blogData));
+
+Article.loadBlogData = function (rawBlogData) {
+  // sort the articles by descending published date
+  rawBlogData.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  });
+  // run each blog article through constructor function and add to the articles array
+  rawBlogData.forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
+}
+
+Article.fetchData = function() {
+  console.log('fetching the data');
+  // check if already in localStorage
+  if (localStorage.blogData) {
+    console.log('Local storage already exists.');
+    Article.loadBlogData(JSON.parse(localStorage.blogData));
+    articleView.loadIndexPage();
+  } else {
+    $.getJSON('data/articleData.json', function(json) {
+      console.log('Local storage does not exist.');
+      localStorage.setItem('blogData', JSON.stringify(json));
+      console.log('Local storage created');
+      Article.loadBlogData(json);
+      articleView.loadIndexPage();
+    });
+  }
+  // QUESTION: not sure if this should go here
+  // articleView.loadIndexPage();
+}
```

Comment:
Yes, it could go there (and be removed from lines 44 and 51). You will need to check that it executes at the correct point, since you're dealing with an asynchronous process. Perhaps it could be added to the end of the loadBlogData function?